VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -180,8 +180,16 @@ public void onRestart() {
             mSelectedItem = -1;
         }
 
-        nextPage = 1;
-        controller.resetSilently();
+        /* Dev Notes:
+        If the activity/fragment needs to be reinstantiated upon restoration,
+        then in some cases the onRestart() callback maybe invoked before view
+        initialization, and thus the controller might not be initialized, and
+        therefore we need to guard this with a null check.
+         */
+        if (controller != null) {
+            nextPage = 1;
```

Comment:
According to [MA-2163][], whenever we return back to this Activity, we silently reload the data to refresh the screen. Usman did open [MA-2205][] to address the concern that this might cause confusion if there is a delay.

[MA-2163]: https://openedx.atlassian.net/browse/MA-2163
[MA-2205]: https://openedx.atlassian.net/browse/MA-2205